condensation-forum/application.py

```diff
@@ -301,6 +314,117 @@ def logoutCallback():
     print("User signed out: " + user["name"], file=sys.stderr)
 
 
+@application.route('/file-delete', methods=['POST'])
+@authManager.requireAuthentication
+def fileManagerDeleteHander():
+    user = authManager.getUserData()
+    fid = int(request.form['file'])
+    
+    if not user:
+        abort(403)
+    id = user['id']
+
+    #delete the file by Cloud_key in AWS S3
+    with dataSessionMgr.session_scope() as dbSession:
+        file1 = query.getFileById(dbSession,fid)
+        file1 = query.extractOutput(file1)
```

Comment:
you should wrap this in a try catch and `flash` a message to the user if it fails

Reply:
Also this deviates from the logic we talked about a few days ago, we only want to delete the database entry if and only if the s3 removal fails

Reply:
Fixed